streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java

```diff
@@ -43,12 +43,11 @@ public ProcessorRecordContext(final long timestamp,
                                   final int partition,
                                   final String topic,
                                   final Headers headers) {
-
         this.timestamp = timestamp;
         this.offset = offset;
         this.topic = topic;
         this.partition = partition;
-        this.headers = headers;
+        this.headers = Objects.requireNonNull(headers);
```

Comment:
I think, for the internal change, this is actually ok. (I am still ok to split this into a separate PR, and keep this on a JavaDocs only PR).

Reply:
Yeah, as long as the tests pass, I think it's ok.